apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -403,383 +402,396 @@ export function buildMachine(
-            invoke: pollPaperStatus(),
-            on: {
-              NO_PAPER_ANYWHERE: 'jam_physically_cleared',
+            resetting_state_machine_no_delay: {
+              entry: ['resetContext', 'endCardlessVoterAuth'],
+              always: 'not_accepting_paper',
             },
-          },
-          jam_physically_cleared: {
-            invoke: {
-              id: 'resetScanAndDriver',
-              src: (context) => {
-                // Issues `reset scan` command, creates a new WebUSBDevice, and reconnects
-                return resetAndReconnect(context.driver);
+            resetting_state_machine_after_jam: {
+              entry: ['resetContext', 'endCardlessVoterAuth'],
+              after: {
+                [NOTIFICATION_DURATION_MS]: 'not_accepting_paper',
               },
-              onDone: {
-                target: 'resetting_state_machine_after_jam',
-                actions: assign({
-                  // Overwrites the old nonfunctional driver in context with the new functional one
-                  driver: (_, event) => event.data,
-                }),
-              },
-            },
-          },
-          resetting_state_machine_after_jam: {
-            entry: async (context) => {
-              await auth.endCardlessVoterSession(
-                constructAuthMachineState(context.workspace)
-              );
-
-              assign({
-                interpretation: undefined,
-                scannedImagePaths: undefined,
-                isPatDeviceConnected: false,
-              });
             },
-            after: {
-              // The frontend needs time to idle in this state so the user can read the status message
-              [RESET_AFTER_JAM_DELAY_MS]: 'not_accepting_paper',
+            resetting_state_machine_after_success: {
+              entry: ['resetContext', 'endCardlessVoterAuth'],
+              always: [
+                {
+                  target: 'empty_ballot_box',
+                  cond: (context) =>
+                    context.workspace.store.getBallotsCastSinceLastBoxChange() >=
+                    MAX_BALLOT_BOX_CAPACITY,
+                },
+                { target: 'not_accepting_paper' },
+              ],
             },
-          },
-          resetting_state_machine_after_success: {
-            entry: async (context) => {
-              await auth.endCardlessVoterSession(
-                constructAuthMachineState(context.workspace)
-              );
-
-              assign({
-                interpretation: undefined,
-                scannedImagePaths: undefined,
-                isPatDeviceConnected: false,
-              });
+            poll_worker_auth_ended_unexpectedly: {
+              entry: ['resetContext', 'endCardlessVoterAuth'],
+              after: {
+                // The frontend needs time to idle in this state so the user can read the status message
+                [NOTIFICATION_DURATION_MS]: 'not_accepting_paper',
```

Comment:
Watching the video, it feels like the paper should be ejected while this message is showing, rather than after. Though I think it's totally fine to leave as is since this is likely an uncommon case.

Reply:
Good point. It's a quick change so not a problem to add even if it's uncommon

[8b2a91a](https://github.com/votingworks/vxsuite/pull/4647/commits/8b2a91a33d4702ac7ef4fa9f2723a8aab9f9376d)